templates/master/00-master/two-node-with-fencing/files/fencing-validator.yaml

```diff
@@ -0,0 +1,502 @@
+        fi
+        sleep 5
+      done
+      err "Timeout waiting for $n NotReady"
+      return 1
+    }
+    
+    wait_ready(){
+      local n="$1" deadline=$((SECONDS+TIMEOUT))
+      log "Waiting for '$n' Ready (API)…"
+      while (( SECONDS < deadline )); do
+        node_ready "$n" && { ok "$n Ready (API)"; break; }
+        sleep 5
+      done
+      (( SECONDS < deadline )) || { err "Timeout waiting Ready for $n"; return 1; }
+      deadline=$((SECONDS+TIMEOUT))
+      log "Waiting for '$n' ONLINE (Pacemaker)…"
+      while (( SECONDS < deadline )); do
+        pcmkonline "$n" && { ok "$n ONLINE (Pacemaker)"; return 0; }
+        sleep 5
+      done
+      err "Timeout waiting Pacemaker ONLINE for $n"; return 1
+    }
+    
+    stonith_show(){
+      host_run "$CONDUCTOR" \
+        "(pcs stonith config || pcs stonith status || pcs stonith show) 2>&1" \
+        || true
+    }
+    
+    check_stonith(){
+      log "Checking STONITH…"
+      local out; out="$(stonith_show)"
+      [[ -n "$out" ]] || { err "No STONITH devices detected (pcs returned empty output)"; return 1; }
+      host_run "$CONDUCTOR" \
+        'pcs property config stonith-enabled 2>&1 || pcs property list stonith-enabled 2>&1 || pcs property show --all stonith-enabled 2>&1' \
+      | grep -Eqi 'stonith-enabled[^[:alnum:]]*true' \
+        || { err "stonith-enabled=false (or not reported)"; return 1; }
+    
+      ok "STONITH present and enabled"
+    }
+    
+    check_daemon_status(){
+      log "Checking Pacemaker daemon status…"
+      local out ds ok_all=1
+      out="$(host_run "$CONDUCTOR" "pcs status --full 2>/dev/null || pcs status 2>/dev/null" 2>/dev/null || true)"
+      ds="$(awk '/^Daemon Status:/,0{print}' <<<"$out")"
+      [[ -n "$ds" ]] || { err "Daemon Status section not found in 'pcs status' output"; return 1; }
+    
+      for svc in corosync pacemaker pcsd; do
+        if ! grep -qiE "^[[:space:]]*$svc:[[:space:]]*(active|running)" <<<"$ds"; then
+          err "Daemon '$svc' not active/enabled"
+          ok_all=0
+        fi
+      done
+    
+      if (( ok_all == 1 )); then
+        ok "Daemon Status: corosync, pacemaker, pcsd active/enabled"
+      else
+        return 1
+      fi
+    }
+    
+    # etcd: need 2 started, non-learner voters
+    node_exec_target(){
+      if [[ "$TRANSPORT" == "ssh" ]]; then
+        [[ "$1" == "$NODE_A" ]] && echo "$IP_A" || echo "$IP_B"
+      else
+        echo "$1"
+      fi
+    }
+    
+    etcd_two_started() {
+      local tgt="$1" out rc
+      out="$(host_run "$tgt" "podman exec etcd sh -lc 'ETCDCTL_API=3 etcdctl member list -w table'" 2>&1)"; rc=$?
+      if (( rc != 0 )); then
+        if grep -Eqi 'no such container|container state improper|not running|missing required container_id' <<<"$out"; then
+          return 1
+        fi
+        err "etcdctl failed on $tgt: ${out##*$'\n'}"
+        return 2
+      fi
+    
+      for ip in "$IP_A" "$IP_B"; do
+        awk -F'|' -v ip="$ip" '
```

Comment:
If we have the output as json, we should be able to check if both IPs exist with jq here. wdyt?